llvm/lib/Target/AIE/AIEHazardRecognizer.cpp

```diff
@@ -392,31 +404,45 @@ auto toHazardType(bool Conflict) {
 // recognizing alternatives
 ScheduleHazardRecognizer::HazardType
 AIEHazardRecognizer::getHazardType(const MCInstrDesc &Desc,
+                                   MemoryBankBits MemoryBanks,
                                    const int DeltaCycles) {
   return toHazardType(checkConflict(
       Scoreboard, ItinData, Desc.getSchedClass(),
       getSlotSet(Desc, *TII->getFormatInterface(), IgnoreUnknownSlotSets),
-      DeltaCycles, FUDepthLimit));
+      MemoryBanks, TII->getFirstMemoryCycle(Desc.getSchedClass()), DeltaCycles,
+      FUDepthLimit));
 }
 
 ScheduleHazardRecognizer::HazardType
 AIEHazardRecognizer::getHazardType(unsigned SchedClass, SlotBits SlotSet,
+                                   MemoryBankBits MemoryBanks,
                                    int DeltaCycles) {
-  return toHazardType(checkConflict(Scoreboard, ItinData, SchedClass, SlotSet,
-                                    DeltaCycles, FUDepthLimit));
+  return toHazardType(checkConflict(
+      Scoreboard, ItinData, SchedClass, SlotSet, MemoryBanks,
+      TII->getFirstMemoryCycle(SchedClass), DeltaCycles, FUDepthLimit));
 }
 
 bool AIEHazardRecognizer::checkConflict(
     const ResourceScoreboard<FuncUnitWrapper> &Scoreboard,
     const InstrItineraryData *ItinData, unsigned SchedClass, SlotBits SlotSet,
+    MemoryBankBits MemoryBanks, std::optional<int> MemoryAccessCycle,
     int DeltaCycles, std::optional<int> FUDepthLimit) {
   assert(Scoreboard.isValidDelta(DeltaCycles));
 
-  // Verify format hazards
+  // Verify slot hazards
```

Comment:
I think format hazards was better. We don't just check that the slots are free, we also check that a format can accommodate all the new slots.